### PR TITLE
Fix errata re-migration when it's been copied to a new repo in Pulp2

### PR DESCRIPTION
**Pull request by goosemania**
closes #8417
https://pulp.plan.io/issues/8417

Required PR:pulp/pulp-2to3-migration-test-fixtures#5

### CODE REVIEW

**Review by dralley — APPROVED** [review]

**Review by dralley — APPROVED** [review]

**Comment by goosemania** [comment]
Tests pass locally, maybe the snapshot is not picked up with the Required PR. Maybe let's merge/review the snapshot first?

**Comment by dralley** [comment]
@goosemania Looks OK, I'm assuming the errors are due to the fixture image

**Comment by jlsherrill** [comment]
This solves the issue for pulp, but exposes another issue.  Katello during reimport was only requesting migrated pulp2 content (of type errata) that were updated since the last import:

/pulp/api/v3/pulp2content/?limit=2000&offset=0&pulp2_content_type_id=erratum&pulp2_last_updated__gt=1616202349

However, now this seems to return nothing after copying existing errata to a new repo and re-migrating.  Looking at the api a little further:

GET /pulp/api/v3/pulp2content/?offset=0&pulp2_content_type_id=erratum&pulp2_id=fff42996-0254-468f-af08-2e86f6f93890 

```
{
  "count": 3,
  "next": null,
  "previous": null,
  "results": [
    {
      "pulp_href": "/pulp/api/v3/pulp2content/22ce7c6b-6e33-402a-ac88-8cd168190a14/",
      "pulp_created": "2021-03-20T01:31:22.628439Z",
      "pulp2_id": "fff42996-0254-468f-af08-2e86f6f93890",
      "pulp2_content_type_id": "erratum",
      "pulp2_last_updated": 1616179795,
      "pulp2_storage_path": null,
      "downloaded": false,
      "pulp3_content": "/pulp/api/v3/content/rpm/advisories/c856e315-4d9d-4b27-8cfe-73f9df024679/",
      "pulp3_repository_version": "/pulp/api/v3/repositories/rpm/rpm/c9655b72-cbd4-4c67-a808-5e8eb092d998/versions/1/"
    },
    {
      "pulp_href": "/pulp/api/v3/pulp2content/59eec133-d2d9-4492-99c4-5399d3519247/",
      "pulp_created": "2021-03-20T01:06:23.873820Z",
      "pulp2_id": "fff42996-0254-468f-af08-2e86f6f93890",
      "pulp2_content_type_id": "erratum",
      "pulp2_last_updated": 1616179795,
      "pulp2_storage_path": null,
      "downloaded": false,
      "pulp3_content": "/pulp/api/v3/content/rpm/advisories/c856e315-4d9d-4b27-8cfe-73f9df024679/",
      "pulp3_repository_version": "/pulp/api/v3/repositories/rpm/rpm/41ac06b7-5d61-4e3c-8584-5afce501ffb5/versions/2/"
    },
    {
      "pulp_href": "/pulp/api/v3/pulp2content/90edfcf8-5828-480b-90cd-318b7aedc251/",
      "pulp_created": "2021-03-19T21:02:30.478957Z",
      "pulp2_id": "fff42996-0254-468f-af08-2e86f6f93890",
      "pulp2_content_type_id": "erratum",
      "pulp2_last_updated": 1616179795,
      "pulp2_storage_path": null,
      "downloaded": false,
      "pulp3_content": "/pulp/api/v3/content/rpm/advisories/c856e315-4d9d-4b27-8cfe-73f9df024679/",
      "pulp3_repository_version": "/pulp/api/v3/repositories/rpm/rpm/9eb340f5-e188-43dc-8dec-d9788c7e4bfc/versions/1/"
    }
  ]
}

```

Notice that the pulp2_last_updated is still the old value and all the same.  This was working previously in that newly copied errata would get a newer pulp2_last_updated timestamp.  This allowed us to not have to reimport all errata which could be very expensive.  Is this fixable?  I'm not sure another way to avoid having to fetch all errata every time.

**Comment by goosemania** [comment]
@jlsherrill , I believe it's pulp 2 copy behaviour. Are you sure it worked before with copy and not with 2 syncs of the same content into different repositories? Migration plugin doesn't modify pulp 2 dates, and I just returned back (with slight modifications) code which was catching errata for which date didn't change properly.

Option 1 is to fix it in pulp 2, however it will make many more publishes operational - the moment you create a new CV, all other ones which contain the same erratum will have operational publish. And I believe it's undesirable.

Option 2 is to try to mimic this using migration plugin. I'll take a look what date can I set it to for your logic to work and not to break any of our optimizations. Maybe to the last updated since last time.

**Comment by goosemania** [comment]
I updated this PR to set the `pulp2_last_updated` date to the time when a unit was copied.
@jlsherrill , I think this meets your needs now. 
I still do not know how it worked in the past (the change of timestamp during copy), I believe it did not.

File: pulp_2to3_migration/app/pre_migration.py [hunk 1]
```diff
@@ -259,8 +259,8 @@ def pre_migrate_content_type(content_model, mutable_type, lazy_type, premigrate_
         ).select_related(
             'pulp2_repository'
         ).only(
-            'pulp2_repository'
-        )
+            'pulp2_repository', 'pulp2_created',
+        ).order_by('pulp2_created')
```
Comment (dralley) [inline pulp_2to3_migration/app/pre_migration.py:263]: This might be painful.

**Review by dralley — APPROVED** [review]

**Pull request merged**